// Protos for random optimization.

syntax = "proto2";

package phd.experimental.compilers.random_opt;

message RandomOptStep {
  optional string input_src = 1;
  optional string opt_pass = 2;
  optional string opt_stderr = 3;
  repeated int32 runtime_ms = 4;
  optional int64 start_time_epoch_ms = 5;
}

message RandomOptRun {
  repeated RandomOptStep step = 1;
}

message RandomOptExperiment {
  repeated string input_src = 1;
  repeated string binary_arg = 2;
  optional int32 num_steps = 3;

  optional RandomOptRun results = 15;
}
